.github/workflows/build-test.yaml

```
@@ -1532,9 +1551,213 @@ jobs:
           ./bin/kots admin-console push-images ./hack/tests/small.airgap ttl.sh/automated-${{ github.run_id }}
 
 
+  validate-kots-admin-console-generate-manifests:
+    runs-on: ubuntu-20.04
+    needs: [ enable-tests, can-run-ci, build-kots, build-kurl-proxy, build-migrations, push-minio, push-mc, push-rqlite ]
+    strategy:
+      fail-fast: false
+      matrix:
+        k8s_version: [ v1.26.1-k3s1 ]
+    env:
+      APP_SLUG: minimal-rbac
+      BASE_KOTS_VERSION: v1.72.0
+      NAMESPACE: generate-manifests
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      
+      - name: download kots binary
+        uses: actions/download-artifact@v2
+        with:
+          name: kots
+          path: bin/
+      
+      - run: chmod +x bin/kots
+      
+      - name: run kots admin-console generate-manifests without k8s context
+        run: |
+          set +e
+          ./bin/kots admin-console generate-manifests -n $NAMESPACE --shared-password password
+      
+      - name: validate that ./admin-console exists and is not empty
+        run: |
+          set +e
+          if [ ! -d ./admin-console ]; then
+            echo "admin-console directory does not exist"
+            exit -1
+          fi
+          if [ -z "$(ls -A ./admin-console)" ]; then
+            echo "admin-console directory is empty"
+            exit -1
+          fi
+
+      - name: remove admin-console directory
+        run: rm -rf ./admin-console
+      
+      - name: install k3s
+        uses: replicatedhq/action-k3s@main
+        with:
+          version: ${{ matrix.k8s_version }}
+          arguments: ${{ startsWith(matrix.k8s_version, 'v1.22') && '--egress-selector-mode disabled' || '' }}
+      
+      - name: download base kots version
+        run: |
+          curl -LO https://github.com/replicatedhq/kots/releases/download/$BASE_KOTS_VERSION/kots_linux_amd64.tar.gz \
+            && tar zxvf kots_linux_amd64.tar.gz \
+            && mv kots kots-$BASE_KOTS_VERSION
+      
+      - name: create namespace
+        run: |
+          set +e
+          kubectl create namespace $NAMESPACE
+      
+      - name: run kots admin-console generate-manifests using base kots version with k8s context
+        run: |
+          set +e
+          ./kots-$BASE_KOTS_VERSION admin-console generate-manifests -n $NAMESPACE --shared-password password
+      
+      - name: apply the generated manifests
+        run: |
+          set +e
+          kubectl apply -f ./admin-console -n $NAMESPACE
+
+      - name: wait for the kotsadm-minio-0 pod to be created
+        run: |
+          set +e
+          COUNTER=1
+          while [ -z "$(kubectl get pods -n $NAMESPACE | grep kotsadm-minio-0)" ]; do
```

nit pick, it can be written this way
```suggestion
          while ! kubectl get pods -n $NAMESPACE | grep -q kotsadm-minio-0; do
```